ToolsInstaller.targets

```diff
@@ -0,0 +1,24 @@
+<Project>
+  <PropertyGroup>
+    <InstallTools Condition=" '$(InstallTools)' == ''">false</InstallTools>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(InstallTools)">
+    <!-- By default, NuGet will not include files starting with '.' or ending with '.nupkg'. Since we are including try-convert via dotnet tools, some of the files fit into this. -->
+    <NoDefaultExcludes>true</NoDefaultExcludes>
+    <TryConvertVersion>0.7.212201</TryConvertVersion>
+    <ToolsDirectory>$(MSBuildProjectDirectory)\tools\</ToolsDirectory>
+  </PropertyGroup>
+
+  <ItemGroup Condition="$(InstallTools)">
+    <None Include="$(ToolsDirectory)**\*">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+    <Content Include="$(ToolsDirectory)**\*" />
+  </ItemGroup>
+  
+  <Target Name="RestoreTools" BeforeTargets="Build" Condition="$(InstallTools) AND !Exists($(ToolsDirectory))">
+    <Exec Command="dotnet tool install try-convert --version $(TryConvertVersion) --tool-path $(ToolsDirectory)" />
```

Comment:
Is it possible to use `dotnet tool restore` instead (with a checked-in tools manifest)? If that works, it would mean that our dev machines would auto-update the local try-convert version when we rev'd it in the manifest.

If that doesn't work, though, this is probably good enough. We can just manually delete try-convert when it needs updated.

Reply:
I tried that originally and there doesn't seem a way to restore to a tool-path